src/lists.py

```diff
@@ -113,7 +117,31 @@ def reverse(x: DLList[T]) -> None:
     >>> print(x)
     [5, 4, 3, 2, 1]
     """
-    ...
+    link = x.head.next
+    end = x.head.prev
+    while link is not end:
+            insert_after(end,link.val)   #not the most efficient
+            remove_link(link)            # as we add one and delete one 
+            link = link.next             # but for now it works   
+
+
+
+def swap(link,next) -> None:
+    insert_after(link,next.val)
+    insert_after(next,link.val)   
+    remove_link(link)
+    remove_link(next)
+    link= link.next    
```

Comment:
This doesn't do anything. You update a local variable and then leave the function.